packages/webclient/src/AzureRestClient.ts

```diff
@@ -17,41 +17,37 @@ export class ApiResult {
 
 export class AzureError {
     public code: any;
-    public message: string | undefined;
-    public statusCode: number | undefined;
+    public message?: string;
+    public statusCode?: number;
     public details: any;
 }
 
 export interface ApiCallback {
     (error: any, result?: any, request?: any, response?: any): void
 }
 
-export function ToError(response: webClient.WebResponse): AzureError {
-    var error = new AzureError();
+export function ToError(response: WebResponse): AzureError {
+    let error = new AzureError();
     error.statusCode = response.statusCode;
     error.message = response.body
+
     if (response.body && response.body.error) {
         error.code = response.body.error.code;
         error.message = response.body.error.message;
         error.details = response.body.error.details;
 
-        console.log("##[error]" + error.message);
+        console.log(`##[error] error.message`);
     }
 
     return error;
 }
 
 export class ServiceClient {
-    private endpoint: IAuthorizationHandler;
-    protected baseUrl: string;
-    protected longRunningOperationRetryTimeout: number;
-
-    public subscriptionId: string;
-
-    constructor(endpoint: IAuthorizationHandler, timeout?: number) {
-        this.endpoint = endpoint;
-        this.subscriptionId = this.endpoint.subscriptionID;
-        this.baseUrl = this.endpoint.baseUrl;
+    constructor(handler: IAuthorizationHandler, timeout?: number) {
+        this._webClient = new WebClient();
+        this._handler = handler;

```

Comment:
should we call this property enpoint only. we as consumers have more context to endpoint.

Reply:
A handler should return a handler, not an endpoint. Since this is a generic library, imo we should not use AzDevOps related terminologies.

Reply:
this is azure specific file. As we discussed should we change it to authorizer?